core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/SwapMove.java

```diff
@@ -44,6 +44,9 @@ public Object getRightEntity() {
 
     @Override
     public boolean isMoveDoable(ScoreDirector<Solution_> scoreDirector) {
+        if (leftEntity == rightEntity) {
+            return false;
+        }
```

Comment:
Ideally, this would have been solved by not generating the move.
Unfortunately, if there is only one entity, that would lead to an endless loop in the move selector - and somehow fixing that would have broken backwards compatibility.